pkg/clusteragent/autoscaling/store.go

```diff
@@ -98,6 +116,26 @@ func (s *Store[T]) GetFiltered(filter func(T) bool) []T {
 	return objects
 }
 
+// GetFilteredByIndexKey returns a copy of all store values matched by the index key
+func (s *Store[T]) GetFilteredByIndexKey(indexKey any) []T {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	ids := s.indexer.GetIDs(indexKey)
+	if len(ids) == 0 {
+		return nil
+	}
+
+	objects := make([]T, 0, len(ids))
+	for _, id := range ids {
+		if obj, exists := s.store[id]; exists {
+			objects = append(objects, obj)
+		}
+	}
+
+	return objects
+}
+
 // Update updates all objects in the store with the result of the `updator` function.
 // Updator func is expected to return the new object and a boolean indicating if the object has changed.
 // The object is updated only if boolean is true, observers are notified only for updated objects after all objects have been updated.
```

Comment:
I'm wondering if also the Update() function must be modified.
From what I understood we use the ownerreference to generate they indexKey. However the owner-reference contains the APIGroup information, and I'm sure how it should work when an object is move to a different apigroup version.